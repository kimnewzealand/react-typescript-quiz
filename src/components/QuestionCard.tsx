import React from 'react'

import { AnswerObject } from '../App'

// Styles
import { Wrapper } from '../App.styles'

type Props = {
    question: string;
    answers: string[];
    callback: (e: React.MouseEvent<HTMLButtonElement>) => void;
    userAnswer: AnswerObject | undefined;
    questionNr: number;
    totalQuestions: number;
}

const QuestionCard: React.FC<Props> = ({ question, answers, callback, userAnswer, questionNr, totalQuestions, children }) => (
    <>
        <div>
            <Wrapper>
                <p className="number">Question: {questionNr}/{totalQuestions}</p>
                <p dangerouslySetInnerHTML={{ __html: question }} />
            </Wrapper>
            <div>
                {answers.map(answer => (
                    <div key={answer}>
                        <button disabled={userAnswer ? true : false} value={answer} onClick={callback}>
                            <span dangerouslySetInnerHTML={{ __html: answer }} />
                        </button>
                    </div>
                ))}
            </div>

        </div>
    </>
)
export default QuestionCard