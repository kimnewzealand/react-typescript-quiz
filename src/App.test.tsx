import React from 'react';
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom'
// Components
import App from './App'
import QuestionCard from './components/QuestionCard'

it('takes a snapshot', () => {
    const { asFragment } = render(<App />)
    expect(asFragment).toMatchSnapshot()
})

it('renders the correct title in the document', () => {
    render(<App />)
    expect(screen.getByText('React Quiz')).toBeInTheDocument();
})

it('should have have a start button', () => {
    render(<App />)
    screen.getByRole('button', { name: /Start/i })
})

it('should have have a next button', () => {
    render(<App />)
    screen.getByRole('button', { name: /Next/i })
});

it('clicks the Start button to render a score of 0', () => {
    render(<App />)
    // Click button
    userEvent.click(screen.getByRole('button', { name: /Start/i }))
    expect(screen.getByText('Score: 0')).toBeInTheDocument()
})

it('clicks the Start button to render first question', () => {
    render(<App />)
    // Click button
    userEvent.click(screen.getByRole('button', { name: /Start/i }))
    render(<QuestionCard questionNr={1}
        totalQuestions={10}
        question={'What does GHz stand for?'}
        answers={['Gigahertz', 'Gegahertz', 'Gagahertz', 'Gugahertz']}
        userAnswer={undefined}
        callback={undefined} />)
    expect(screen.getByText('Question: 1/10')).toBeInTheDocument()
})

it('clicks the Next button to render second question', () => {
    render(<App />)
    // Click button
    userEvent.click(screen.getByRole('button', { name: /Start/i }))
    render(<QuestionCard questionNr={1}
        totalQuestions={10}
        question={'What does GHz stand for?'}
        answers={['Gigahertz', 'Gegahertz', 'Gagahertz', 'Gugahertz']}
        userAnswer={undefined}
        callback={undefined} />)
    userEvent.click(screen.getByRole('button', { name: /Next/i }))
    render(<QuestionCard questionNr={2}
        totalQuestions={10}
        question={'What does MHz stand for?'}
        answers={['Megahertz', 'Migahertz', 'Magahertz', 'Mugahertz']}
        userAnswer={undefined}
        callback={undefined} />)
    expect(screen.getByText('Question: 2/10')).toBeInTheDocument()
})

