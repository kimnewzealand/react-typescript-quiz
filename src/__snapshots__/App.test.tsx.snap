// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`takes a snapshot 1`] = `[Function]`;
